runtime/lua/vim/lsp.lua

```diff
@@ -1156,6 +1156,12 @@ function lsp.buf_attach_client(bufnr, client_id)
     client_id = {client_id, 'n'};
   }
   bufnr = resolve_bufnr(bufnr)
+  if not vim.api.nvim_buf_is_loaded(bufnr) then
+    local _ = log.warn() and log.warn(
+      string.format("buf_attach_client called on unloaded buffer (id: %d): ", bufnr)
+    )
+    return
```

Comment:
Since the function returns `true` on success do you want to make this return `false` on failure to keep the return type consistent?

Reply:
I thought about it but I think conventionally we return nil on failure (although this is a loose convention). Most people shouldn't be explicitly checking false anyways?